src/vconsole/vconsole-setup.c

```diff
@@ -54,6 +54,55 @@ static bool is_vconsole(int fd) {
         return ioctl(fd, TIOCLINUX, data) >= 0;
 }
 
+static bool is_allocated(int fd, int idx) {
+        struct vt_stat vcs = {};
+        char vcname[strlen("/dev/vcs") + DECIMAL_STR_MAX(int)];
+
+        if (fd >= 0) {
+                if (ioctl(fd, VT_GETSTATE, &vcs) < 0) {
+                        (void) log_warning_errno(errno, "VT_GETSTATE failed: %m");
+                        return false;
+                }
+                idx = vcs.v_active;
+        }
+
+        xsprintf(vcname, "/dev/vcs%i", idx);
+        if (access(vcname, F_OK) < 0)
+                return false;
+        return true;
+}
+
+static bool is_settable(int *fd, int idx) {
```

Comment:
Those two functions are used after following patch https://github.com/systemd/systemd/pull/3742/commits/813f1eb5d002e5d117f6709fc19b2546b42724a4

Essentially if fd is set, fd is used. Otherwise idx is used, but that's done later in per-console function.